lib/msf/core/post/windows/services.rb

```diff
@@ -229,23 +260,40 @@ def service_list
   def service_info(name)
     service = {}
 
-    if load_extapi
-      begin
-        return session.extapi.service.query(name)
-      rescue Rex::Post::Meterpreter::RequestError => e
-        vprint_error("Request Error #{e} falling back to registry technique")
+    case session.type
+    when "powershell"
+      if SERVICES.empty?
```

Comment:
Yeah, usually the old way it has to enumerate each service in registry. In powershell you can enumerate this with one command as i've done in list_services